### PR TITLE
Exit kbfs when service shuts down, even with no fs mount

### PR DESCRIPTION
**Pull request by zanderz**
so it can be more like other platforms, for cleaner shutdown and updating.
The separate NotifyServiceHandler seemed necessary because there already was a Shutdown() (no argument) satisfying some other interface.
Related to DESKTOP-5956, which will be blocked until this is merged.
Tested in a few production builds today.
@taruti @maxtaco @jzila

### CODE REVIEW

**Comment by zanderz** [comment]
I realize this will close KBFS in prod mode every time the service closes, not just when `keybase ctl stop --kill-kbfs` is issued. If we don't like that, we could forward an argument through the rpc layers to distinguish that case, but I think macos works like this anyway.

File: libkbfs/keybase_daemon_rpc.go [hunk 1]
```diff
@@ -403,9 +413,31 @@ func (k *KeybaseDaemonRPC) Shutdown() {
...
+func (s notifyServiceHandler) Shutdown(_ context.Context) error {
+	s.log.Warning("NotifyService: Shutdown")
+	if runtime.GOOS == "windows" {
+		os.Exit(0)
+	}
+	return nil
+}
+
+// newNotifyServiceHandler makes a new NotifyServiceHandler
+func newNotifyServiceHandler(config Config, log logger.Logger) keybase1.NotifyServiceInterface {
```
Comment (taruti) [inline libkbfs/keybase_daemon_rpc.go]: Is there a reason we use this by value and not by reference here and in the methods?

Comment (zanderz) [reply libkbfs/keybase_daemon_rpc.go]: No, will change

**Comment by zanderz** [comment]
yes, see `watchdog2`. The gui starts the service via the watchdog, and so does the system startup shortcut.

**Comment by zanderz** [comment]
That counts as a thumb I believe?

**Comment by taruti** [comment]
LGTM

**Pull request merged**